### PR TITLE
Add license attribute to all tools

### LINKED ISSUE #92: Include license as a tool attribute
license of the Galaxy tool, or the python code shipped with it can be annotated on the `<tool` level.

E.g. here:
https://github.com/muon-spectroscopy-computational-project/muon-galaxy-tools/blob/bd5632424b9e1ba94d7c47c91f0e7f08c5fda15d/pm_dftb_opt/pm_dftb_opt.xml#L69

See documentation:
https://docs.galaxyproject.org/en/latest/dev/schema.html#tool
https://spdx.org/licenses/CC-BY-SA-4.0.html

### PR DESCRIPTION
**Pull request by patrick-austin**
Add the `license` attribute to all tools. This should now appear alongside citations, requirements etc. at the bottom of the page for each tool.

For `muspinsim...` tools, this was `MIT`. For `pm...` tools, it was `GPL-3.0` but this identifier is deprecated in favour of `GPL-3.0-only`:
https://spdx.org/licenses/GPL-3.0.html
https://spdx.org/licenses/GPL-3.0-only.html

For `cif2cell`, we didn't already have a license recorded in the xml so I checked https://pypi.org/project/cif2cell/

For [pm_dftb_opt](https://github.com/muon-spectroscopy-computational-project/muon-galaxy-tools/compare/patrick/92-license-attribute?expand=1#diff-26f80979afa2a3fc5cb7c5488e673e1dc02a3f73600bbd1cfe1882ee4dc95011) the software/tool itself uses GPL3, but some of the parameter sets that can be used with DFTB use https://spdx.org/licenses/CC-BY-SA-4.0.html
e.g.
https://dftb.org/parameters/download/pbc/pbc-0-3-cc
which requires that we "indicate the Licensed Material is licensed under this Public License, and include the text of, or the URI or hyperlink to, this Public License."
I've therefore left the existing disclaimer in the help, but used GPL3 as the `license` attribute seeing as that applies to the tool itself. If someone with more knowledge of licensing thinks another approach would be more suitable, I'm happy to change this.

Closes #92

### CODE REVIEW

**Comment by elichad** [comment]
I had a look at https://github.com/galaxyproject/tools-iuc and saw it has an overarching MIT license that I assume applies even to tools that wrap more restrictively-licensed software where the `license` tag is undefined. @bgruening have I understood this right?

It seems like it might be better to just apply MIT to everything in this repo (then we can also display that on GitHub) and add a license disclaimer to the help text of the GPL/other-licenced tools? Will mean less worry for future tools

**Comment by bgruening** [comment]
My understanding is that MIT is for the tool-XML files. The license annotation in the XML file might indicate the license of additional scripts associated with the XML tools. 

It might be good to open an issue at https://github.com/galaxy-iuc/standards to discuss this further and enhance our recommendations.

**Comment by elichad** [comment]
Ok, made an issue. Then here I'll stick with my original statement that we may as well use MIT for everything.

File: pm_asephonons/pm_asephonons.xml [hunk 1]
```diff
@@ -110,6 +110,7 @@
 
         Given an input structure, this tool generates a phonon report for that structure using the parameters provided.
 
+        Note that while this tool uses the MIT license, PyMuonSuite itself is distributed under GPLv3.
```
Comment (elichad) [inline pm_asephonons/pm_asephonons.xml:113]:
```suggestion
        PyMuonSuite is distributed under the GPLv3 license. This tool wrapper is distributed under the MIT license.
```

File: cif2cell/cif2cell.xml [hunk 2]
```diff
@@ -40,6 +40,8 @@
 
         Given an input .cif structure file, convert to equivalent .cell structure file.
         Uses electronic structure program 'castep'
+
+        Note that while this tool uses the MIT license, cif2cell itself is distributed under GPLv3.
```
Comment (elichad) [inline cif2cell/cif2cell.xml:44]:
```suggestion
        cif2cell is distributed under the GPLv3 license. This tool wrapper is distributed under the MIT license.
```

File: pm_dftb_opt/pm_dftb_opt.xml [hunk 3]
```diff
@@ -70,6 +70,8 @@
     <help><![CDATA[
         usage: dftb+
         Given some input muonated structures (structures containing a muon), applies DFTB+ optimization to refine them.
+        
+        Note that while this tool uses the MIT license, PyMuonSuite itself is distributed under GPLv3.
```
Comment (elichad) [inline pm_dftb_opt/pm_dftb_opt.xml:74]:
```suggestion
        PyMuonSuite is distributed under the GPLv3 license. This tool wrapper is distributed under the MIT license.
```

File: pm_muairss_read/pm_muairss_read.xml [hunk 4]
```diff
@@ -88,6 +88,8 @@
         'Generate muon structures' tool or pm-muairss.
 
         Command-line usage: pm-muairss [-h] -t r structures parameter_file
+
+        Note that while this tool uses the MIT license, PyMuonSuite itself is distributed under GPLv3.
```
Comment (elichad) [inline pm_muairss_read/pm_muairss_read.xml:92]:
```suggestion
        PyMuonSuite is distributed under the GPLv3 license. This tool wrapper is distributed under the MIT license.
```

File: pm_muairss_write/pm_muairss_write.xml [hunk 5]
```diff
@@ -177,6 +177,8 @@
         .. _pm-muairss documentation: https://github.com/muon-spectroscopy-computational-project/pymuon-suite/blob/master/docs/muairss-gen-docs.md
         .. _YAML config generator: /tool_runner?tool_id=pm_yaml_config
         .. _Generate UEP outputs: /tool_runner?tool_id=pm_uep_opt_write
+
+        Note that while this tool uses the MIT license, PyMuonSuite itself is distributed under GPLv3.
```
Comment (elichad) [inline pm_muairss_write/pm_muairss_write.xml:181]:
```suggestion
        PyMuonSuite is distributed under the GPLv3 license. This tool wrapper is distributed under the MIT license.
```

File: pm_symmetry/pm_symmetry.xml [hunk 6]
```diff
@@ -63,6 +63,8 @@
         usage: pm-symmetry structure
 
         Given an input structure, generates a Wyckoff points symmetry report for that structure.
+        
+        Note that while this tool uses the MIT license, PyMuonSuite itself is distributed under GPLv3.
```
Comment (elichad) [inline pm_symmetry/pm_symmetry.xml:67]:
```suggestion
        PyMuonSuite is distributed under the GPLv3 license. This tool wrapper is distributed under the MIT license.
```

File: pm_uep_opt/pm_uep_opt.xml [hunk 7]
```diff
@@ -106,6 +106,8 @@
         Command-line usage: pm-uep-opt [-h] [-t {r,w}] structures parameter_file
 
         .. _Generate muonated structures: /tool_runner?tool_id=pm_muairss_write
+
+        Note that while this tool uses the MIT license, PyMuonSuite itself is distributed under GPLv3.
```
Comment (elichad) [inline pm_uep_opt/pm_uep_opt.xml:110]:
```suggestion
        PyMuonSuite is distributed under the GPLv3 license. This tool wrapper is distributed under the MIT license.
```

File: pm_yaml_config/pm_yaml_config.xml [hunk 8]
```diff
@@ -215,6 +215,8 @@
     <help><![CDATA[
         Creates a YAML configuration file from input parameters.
 
+        Note that while this tool uses the MIT license, PyMuonSuite itself is distributed under GPLv3.
```
Comment (elichad) [inline pm_yaml_config/pm_yaml_config.xml:218]:
```suggestion
        PyMuonSuite is distributed under the GPLv3 license. This tool wrapper is distributed under the MIT license.
```

**Review by elichad — COMMENTED** [review]
Suggesting a more neutral* license disclaimer. Otherwise this looks fine.

*not sure exactly what word to use, but removing the feel of 'X, but Y'

**Review by elichad — APPROVED** [review]

**Pull request merged**